zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -30,87 +37,69 @@
   private static final String SHARED_PROCESS = "shared_process";
   private String id;
   private String name;
-  private String group;
-  private String description;
+  private String refName; // always be null in case of InterpreterSettingRef
   private Properties properties;
-  private transient InterpreterGroupFactory interpreterGroupFactory;
 
-  // use 'interpreterGroup' as a field name to keep backward compatibility of
-  // conf/interpreter.json file format
-  private List<InterpreterInfo> interpreterGroup;
+  @SerializedName("interpreterGroup")
+  private List<InterpreterInfo> interpreterInfos;
   private transient Map<String, InterpreterGroup> interpreterGroupRef = new HashMap<>();
   private List<Dependency> dependencies;
   private InterpreterOption option;
+  private transient String path;
+
+  @Deprecated
+  private String group;
```

Comment:
If field name is used internally only, `refName` might help understand relation between `InterpreterSetting.name` and `InterpreterSettingRef.name`.

However, field name in `InterpreterSetting` change format of conf/interpreter.json and interpreter setting creation REST api.

So i'm not sure it's good idea to change file format and REST api to improve readability of internal code. Users might already have some external tools and codes that interact with conf/interpreter.json and interpreter setting creation REST api. 

I think we can just add comment in the source code to clarify how InterpreterSettingRef and InterpreterSetting is mapped, without field name change.